### PR TITLE
!!! [TASK] change requirements and dataprocessor

### PR DESCRIPTION
**Pull request by achimfritz**
* drop TYPO3 10 Support
* add TYPO3 13 Support
* replace dataprocessor dataProcessing.1421884800 = TYPO3\CMS\Frontend\DataProcessing\DatabaseQueryProcessor with dataProcessing.1421884800 = B13\Listelements\DataProcessing\ListItemsDataProcessor

### CODE REVIEW

File: Classes/DataProcessing/ListItemsDataProcessor.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function process(
         $processedRecordVariables = [];
         foreach ($items as $key => $record) {
             $recordContentObjectRenderer = GeneralUtility::makeInstance(ContentObjectRenderer::class);
-            $recordContentObjectRenderer->start($record, 'tx_listelements_item', $request);
+            $recordContentObjectRenderer->start($record, 'tx_listelements_item');
```
Comment (bmack) [inline Classes/DataProcessing/ListItemsDataProcessor.php:54]: we then need $recordContentObjectRenderer->setRequest() though then.

File: Configuration/TCA/tx_listelements_item.php [hunk 2]
```diff
@@ -97,10 +78,9 @@
         'sorting_foreign' => [
             'label' => 'LLL:EXT:listelements/Resources/Private/Language/locallang_db.xlf:tx_listelements_item.sorting_foreign',
             'config' => [
-                'type' => 'input',
+                'type' => 'number',
```
Comment (bmack) [inline Configuration/TCA/tx_listelements_item.php:81]: IIRC, "number" was introduced in v12.

File: Configuration/TCA/tx_listelements_item.php [hunk 3]
```diff
@@ -177,118 +157,30 @@
...
-                            ],
-                        ],
-                    ],
-                ]
-                // ,
-                // set this if you need it in your content element configuration (use allowedFileExtensions)
-                // $GLOBALS['TYPO3_CONF_VARS']['GFX']['imagefile_ext']
-            ),
+            'config' => [
+                'type' => 'file',
```
Comment (bmack) [inline Configuration/TCA/tx_listelements_item.php:161]: same here, v12+ only

File: Configuration/TCA/tx_listelements_item.php [hunk 4]
```diff
@@ -177,118 +157,30 @@
...
+                    'createNewRelationLinkTitle' => 'LLL:EXT:listelements/Resources/Private/Language/locallang_db.xlf:tx_listelements_item.assets.addFileReference',
+                ],
+            ],
         ],
         'link' => [
             'label' => 'LLL:EXT:listelements/Resources/Private/Language/locallang_db.xlf:tx_listelements_item.link',
             'config' => [
-                'type' => 'input',
-                'renderType' => 'inputLink',
+                'type' => 'link',
```
Comment (bmack) [inline Configuration/TCA/tx_listelements_item.php:179]: same here.

File: Classes/DataProcessing/ListItemsDataProcessor.php (hunk 1 above)
Comment (achimfritz) [reply Classes/DataProcessing/ListItemsDataProcessor.php:54]: 👍 , done

File: Configuration/TCA/tx_listelements_item.php (hunk 2 above)
Comment (achimfritz) [reply Configuration/TCA/tx_listelements_item.php:81]: will be overriden in https://github.com/b13/listelements/blob/65354bca4f5b17f86987a9166d392a4bacc80803/Configuration/TCA/Overrides/tx_listelements_item.php

File: Configuration/TCA/tx_listelements_item.php (hunk 3 above)
Comment (achimfritz) [reply Configuration/TCA/tx_listelements_item.php:161]: https://github.com/b13/listelements/blob/65354bca4f5b17f86987a9166d392a4bacc80803/Configuration/TCA/Overrides/tx_listelements_item.php

File: Configuration/TCA/tx_listelements_item.php (hunk 4 above)
Comment (achimfritz) [reply Configuration/TCA/tx_listelements_item.php:179]: https://github.com/b13/listelements/blob/65354bca4f5b17f86987a9166d392a4bacc80803/Configuration/TCA/Overrides/tx_listelements_item.php

**Review by bmack — APPROVED** [review]

**Pull request merged**